how-to/workspace-platform-starter/client/src/framework/shapes/favorite-shapes.ts

```diff
@@ -0,0 +1,217 @@
+
+	/**
+	 * Does this favorite have a suggested label that can be used to avoid a lookup
+	 */
+	label?: string;
+
+	/**
+	 * Does this favorite have a suggested icon that can be used to avoid a lookup
+	 */
+	icon?: string;
+}
+
+/**
+ * Info to return to interested parties to help them support favorites
+ */
+export interface FavoriteInfo {
+	/**
+	 * The path to an icon that can be used to indicate the ability to favorite
+	 */
+	favoriteIcon?: string;
+	/**
+	 * The path to an icon that can be used to indicate the ability to remove this favorite
+	 */
+	unfavoriteIcon?: string;
+	/**
+	 * A command that supporting modules should listen for if they are to display favorites that fall under them
+	 */
+	command?: string;
+	/**
+	 * What types of favorite item are supported on this platform
+	 */
+	enabledTypes?: FavoriteTypeNames[];
+	/**
+	 * Is favorite support enabled on this platform.
+	 */
+	isEnabled: boolean;
+}
+
+/**
+ * A client that can be used to provide access to some or all of the favorite functionality
+ */
+export interface FavoriteClient {
+	/**
+	 * The ability to request supporting information about whether favorites are initialized for the platform and supporting information.
+	 * @returns Supporting information.
+	 */
+	getInfo: () => FavoriteInfo;
+	/**
+	 * The ability to request all (or some if by type) of the saved favorites
+	 * @param byType the type of saved favorite you are looking for
+	 * @returns An array of saved favorites or an empty array if it was unable to get any back
+	 */
+	getSavedFavorites: (byType: FavoriteTypeNames) => Promise<FavoriteEntry[]>;
+	/**
+	 * The ability to request a particular saved favorite.
+	 * @param id the id of the favorite you are looking for
+	 * @returns the saved favorite if available or false if it didn't exist
+	 */
+	getSavedFavorite: (id: string) => Promise<FavoriteEntry | undefined>;
+	/**
+	 * The ability to save a favorite.
+	 * @param favorite the Favorite you wish to save
+	 * @returns whether or not the favorite was saved
+	 */
+	setSavedFavorite?: (favorite: FavoriteEntry) => Promise<boolean>;
+	/**
+	 * The ability to remove/delete a saved favorite.
+	 * @param id The id of the favorite to delete
+	 * @returns whether or not the favorite was deleted.
+	 */
+	deleteSavedFavorite?: (id: string) => Promise<boolean>;
+}
+
+/**
+ * A request type for the FavoriteEndpoint that gets all saved favorite entries
+ */
+export interface EndpointFavoriteListRequest {
+	/**
+	 * The id of the platform making the request
+	 */
+	platform: string;
+	/**
+	 * The type if specified should be used to filter the response to only send the entries that are relevant
+	 */
+	favoriteType?: FavoriteTypeNames;
+}
+
+/**
+ * The response after the request for favorites was fulfilled
+ */
+export interface EndpointFavoriteListResponse {
+	/**
+	 * The list of favorite entries with information of what platform versions they were originally saved against
+	 */
+	[key: string]: {
```

Comment:
Should this be an array ? In the future we may have a way of reorganizing favorites where the ordering is critical

Reply:
So this was to match workspaces and pages so if someone had platform storage working it would slot in. Open to discuss.